source/agora/test/Byzantine.d

```diff
@@ -333,3 +378,36 @@ unittest
     assert(network.envelope_type_counts.confirm_count == 0, "The block should not have been confirmed!");
     assert(network.envelope_type_counts.externalize_count == 0, "The block should not have been externalized!");
 }
+
+/// MultiSig test: One node signs with an invalid block signature. Spynode does not sign.
+/// The block should only have 4 / 6 block signatures added
+unittest
+{
+    TestConf conf = { quorum_threshold : 51 };
+    auto network = makeTestNetwork!(ByzantineManager!(true, 0, 0, 1))(conf);
+    network.start();
+    scope(exit) network.shutdown();
+    scope(failure) network.printLogs();
+    network.waitForDiscovery();
+    auto nodes = network.clients;
+    auto spy_node = nodes[$ - 1];
+    assert(spy_node.getQuorumConfig().threshold == 4); // We should need 4 nodes
+    auto txes = genesisSpendable().map!(txb => txb.sign()).array();
+    txes.each!(tx => spy_node.putTransaction(tx));
+    network.setTimeFor(Height(1));  // trigger consensus
+    waitForCount(GenesisValidators - 1, &network.envelope_type_counts.nominate_count, "nominate");
```

Comment:
we cannot rely on nomination counter's please see 
https://github.com/bpfkorea/agora/issues/1431
I am fixing the old cases in my pull request here:
https://github.com/bpfkorea/agora/pull/1437

Reply:
Thanks I created a new test for this that does not rely on the counters.